src/content/docs/en/guides/deploy/space.mdx

```diff
@@ -0,0 +1,109 @@
+
+## Prerequisite
+
+To push an Astro site to Space, make sure you first:
+
+- Create a [Space account](https://deta.space)
+- Install the [Space CLI](https://deta.space/docs/en/basics/cli) and log in
+
+No matter the type of Astro site you are building, create a Space project **inside** the directory of your Astro project. Run the CLI and follow the instructions on the screen.    
+
+```bash
+space new
+```
+
+:::note
+The Space CLI might prompt you with the `nodejs16` engine configuration. You may accept or disregard the recommendation, we will guide you through both options.
+:::
+
+
+## Static site
+
+If you build a simple Astro site with no adapters, you don’t need any extra configuration to your Astro project. 
+
+Edit the `Spacefile` file generated by the Space CLI on the root of your project. You will configure the `nodejs16` engine, add the `build` command, and serve the `dist` directory generated by Astro. 
+
+```yaml title="Spacefile"
+# Spacefile Docs: https://go.deta.dev/docs/spacefile/v0
+v: 0
+micros:
+  - name: static-astro-in-space
+    src: .
+    engine: static
+    commands:
+      - npm run build
+    serve: dist
+```
+
+## Server-side rendering
+
+If you wish to use [server-side rendering (SSR)](https://docs.astro.build/en/guides/server-side-rendering/) in your Astro project, you will need to install the [`@astrojs/node`](https://docs.astro.build/en/guides/integrations-guide/node/) adapter first. 
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+  ```shell
+  npx astro add node
+  ```
+  </Fragment>
+  <Fragment slot="yarn">
+  ```shell
+  yarn astro add node
+  ```
+  </Fragment>
+  <Fragment slot="pnpm">
+  ```shell
+  pnpm astro add node
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+If you need to further configure the adapter, please follow the [`@astrojs/node` documentation](https://docs.astro.build/en/guides/integrations-guide/node/).
+
+:::caution
+When migrating from a static site to a server-rendered site, make sure you make all the proper adjustments based on the type of Astro site you are building. There are bits of configuration sprinkled around the documentation worth checking, but it's not limited to:
+
+- [Dynamic routes for SSR](https://docs.astro.build/en/core-concepts/routing/#server-ssr-mode)
+- [Modifying the `[...slug]` example for SSR](https://docs.astro.build/en/core-concepts/routing/#modifying-the-slug-example-for-ssr)
+- [Building the content collections for SSR](https://docs.astro.build/en/guides/content-collections/#building-for-server-output-ssr)
+:::
+
+Edit the `Spacefile` to configure the `nodejs16` engine, add the `build` command, include the `dist` directory generated by Astro, and run the `node` command.
+
+```yaml title="Spacefile"
+# Spacefile Docs: https://go.deta.dev/docs/spacefile/v0
+v: 0
+micros:
+  - name: ssr-astro-in-space
+    src: .
+    engine: nodejs16
+    commands:
+      - npm run build
+    include:
+      - dist
+    run: "node ./dist/server/entry.mjs"
+```
+
```

Comment:
To actually get your Astro app running on Space you need to push your project. Added a new section on how to do that.

````suggestion

## How to deploy

Once you have your project setup the only thing left to do is push your project to Space:

```bash
space push
```

This will run the build process and create a new [Space app instance](https://deta.space/docs/en/basics/revisions#testing-changes) where you can access your Astro app.

By default Space apps are private and are only accessible to you. If you want to make your app available to others you can use [Public Routes](https://deta.space/docs/en/reference/spacefile#public_routes) to make parts of your app public or you can [create a release](https://deta.space/docs/en/basics/releases) to let others install your app into their own personal cloud.

````